main.py

```diff
@@ -0,0 +1,89 @@
+import json
+import os
+import requests
+import re
+
+
+owner, repo_name = os.environ["OWNER_AND_REPO"].split("/")
+
+url = "https://api.github.com/graphql"
+query = """
+query getDiscussions {
+    repository(name: "$repo_name", owner: "$owner") {
+        discussions(first: 100) {
+            nodes {
+                title
+                body
+                author {
+                    login
+                }
+            }
+            totalCount
+        }
+    }
+}
+"""
+
+variables = {
+    "repo_name": repo_name,
+    "owner": owner,
+}
+
+def apply_variables(query, variables):
+    for name, value in variables.items():
+        query = query.replace("$"+name, str(value))
+
+    return query
+
+def parse_discussions(response):
```

Comment:
For constancy’s sake, please also comment the method implementation including passed down arguments. [Documentation strings for python](https://peps.python.org/pep-0008/#documentation-strings)